src/Chapter2.hs

```diff
@@ -500,7 +516,9 @@ True
 >>> isThird42 [42, 42, 0, 42]
 False
 -}
-isThird42 = error "isThird42: Not implemented!"
+isThird42 :: [Int] -> Bool
+isThird42 (_ : _ : 42 : _) = True
+isThird42 _ = False
```

Comment:
@chshersh I had a question about pattern-matching in Haskell. Is pattern Macthing in some way related to Formal Verification methods? Please correct me if I'm way off mark here :) :)

Reply:
@neelchoudhury Pattern matching can be used for formal verification methods. For example, if you model your system with algebraic data types (explained in chapter three), then you can write functions on them using pattern matching. And if all patterns are covered (this is something the compiler can check automatically), you can say that your implementation is total and handles all cases. This is verification in some way. But usually, when people talk about Formal Verification methods, they mean languages like Agda, Idris or Coq that have a more powerful type system with dependent types and that can actually proof various theorems about the code. Haskell is somewhere in the middle. It doesn't have the full complexity (by default) of all the sophisticated type systems, but it already has some features that can decrease the rate of bugs :beetle: in the code :slightly_smiling_face:

Reply:
I see. Thanks for explaining so well !

Reply:
@neelchoudhury You may be interested in Liquid Haskell. It's the closest thing to formal verification in Haskell that I know of.

* https://ucsd-progsys.github.io/liquidhaskell-blog/

Reply:
Thanks :)